packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -1090,11 +1102,44 @@ class PopupMenuButton<T> extends StatefulWidget {
   PopupMenuButtonState<T> createState() => PopupMenuButtonState<T>();
 }
 
+/// Signature for the callback used by [showMenu] to obtain the position of the
+/// [PopupMenuButton].
+///
+/// Used by [showMenu].
+typedef PopupMenuButtonPositionCallback = RelativeRect Function();
+
 /// The [State] for a [PopupMenuButton].
 ///
 /// See [showButtonMenu] for a way to programmatically open the popup menu
 /// of your button state.
 class PopupMenuButtonState<T> extends State<PopupMenuButton<T>> {
+  GlobalKey _menuGlobalKey = GlobalKey();
+  RelativeRect? _buttonPosition;
+
+  RelativeRect _getButtonPosition() => _buttonPosition!;
+
+  RelativeRect _calculateButtonPosition() {
+    final RenderBox button = context.findRenderObject()! as RenderBox;
+    final RenderBox overlay = Navigator.of(context).overlay!.context.findRenderObject()! as RenderBox;
+    return RelativeRect.fromRect(
+      Rect.fromPoints(
+        button.localToGlobal(widget.offset, ancestor: overlay),
+        button.localToGlobal(button.size.bottomRight(Offset.zero) + widget.offset, ancestor: overlay),
+      ),
+      Offset.zero & overlay.size,
+    );
+  }
+
+  void _maybeUpdateMenuPosition() {
+    WidgetsBinding.instance!.addPostFrameCallback((Duration duration) {
+      final RelativeRect newPosition = _calculateButtonPosition();
+      if (newPosition != _buttonPosition) {
+        _menuGlobalKey.currentState?.setState(() {});
+        _buttonPosition = newPosition;
```

Comment:
I think this should be:
```suggestion
        _menuGlobalKey.currentState?.setState(() {
          _buttonPosition = newPosition;
        });
```

Reply:
haha, we need to update the value even the menu does not open.

Reply:
Sorry about that, I misread this. Completely missed that the menu and the button are two separate widgets